File: DayMap V2/mac/DayMap/CascadingVerticalOnlyScrollView.m
//
//  CascadingVerticalOnlyScrollView.m
//  DayMap
//
//  Created by Chad Seldomridge on 3/8/11.
//  Copyright 2012 Whetstone Apps, LLC. All rights reserved.
//

#import "CascadingVerticalOnlyScrollView.h"

@implementation CascadingVerticalOnlyScrollView

//- (BOOL)isCompatibleWithResponsiveScrolling {
//	return NO;
//}

- (void)scrollWheel:(NSEvent *)event
{
//	NSLog(@"dX = %f, dY = %f", event.scrollingDeltaX, event.scrollingDeltaY);
	if (fabs(event.scrollingDeltaX) > fabs(event.scrollingDeltaY)) {
		[[self nextResponder] scrollWheel:event];
	}
	else {
		[super scrollWheel:event];
	}
}

@end

File: DayMap/CascadingVerticalOnlyScrollView.swift
//
//  CascadingVerticalOnlyScrollView.swift
//  DayMap
//
//  Created by Chad Seldomridge on 11/6/17.
//  Copyright © 2017 Whetstone Apps. All rights reserved.
//

import Cocoa

class CascadingVerticalOnlyScrollView: NSScrollView {
	override func scrollWheel(with event: NSEvent) {
		if abs(event.scrollingDeltaX) > abs(event.scrollingDeltaY) {
			self.nextResponder?.scrollWheel(with: event)
			}
		else {
			super.scrollWheel(with: event)
		}
	}
}
